packages/waku/src/lib/utils/stream.ts

```diff
@@ -11,6 +11,31 @@ export const concatUint8Arrays = (arrs: Uint8Array[]): Uint8Array => {
   return array;
 };
 
+export const streamToArrayBuffer = async (stream: ReadableStream) => {
+  const reader = stream.getReader();
+  const chunks = [];
+  let totalSize = 0;
+  let done = false;
+  let value: Uint8Array | undefined;
+
+  do {
+    ({ done, value } = await reader.read());
+    if (!done && value) {
+      chunks.push(value);
+      totalSize += value.length;
+    }
+  } while (!done);
+
+  const result = new Uint8Array(totalSize);
```

Comment:
A little concern for the performance is the two loops, but it's easier to know the `totalSize`.

Reply:
yeah may try to remove the second loop, theres surely a better way

Reply:
left for now.. according to ai the performance shouldn't be big concern

Reply:
I don't trust ai in this case. two loops should be of impact. This is fine for now. Can you leave a FIXME comment in the code?